backend/app/api/tests/e2e/charge-members.test.ts

```diff
@@ -0,0 +1,191 @@
+import { Request, Response } from '@simonbackx/simple-endpoints';
+import { MemberFactory, Organization, OrganizationFactory, RegistrationPeriod, RegistrationPeriodFactory, Token, User, UserFactory } from '@stamhoofd/models';
+import { BalanceItemWithPayments, ChargeMembersRequest, LimitedFilteredRequest, PermissionLevel, Permissions, StamhoofdFilter, Version } from '@stamhoofd/structures';
+import { ChargeMembersEndpoint } from '../../src/endpoints/admin/members/ChargeMembersEndpoint';
+import { GetMemberBalanceEndpoint } from '../../src/endpoints/organization/dashboard/payments/GetMemberBalanceEndpoint';
+import { testServer } from '../helpers/TestServer';
+
+describe('E2E.ChargeMembers', () => {
+    const chargeMembersEndpoint = new ChargeMembersEndpoint();
+    const memberBalanceEndpoint = new GetMemberBalanceEndpoint();
+    let period: RegistrationPeriod;
+    let organization: Organization;
+    let admin: User;
+    let adminToken: Token;
+
+    const postCharge = async (filter: StamhoofdFilter, body: ChargeMembersRequest, token: Token) => {
+        const request = Request.buildJson('POST', `/v${Version}/admin/charge-members`, undefined, body);
+        const filterRequest = new LimitedFilteredRequest({
+            filter,
+            limit: 100,
+        });
+        request.query = filterRequest.encode({ version: Version }) as any;
+        request.headers.authorization = 'Bearer ' + token.accessToken;
+        return await testServer.test(chargeMembersEndpoint, request);
+    };
+
+    const getBalance = async (memberId: string, organization: Organization, token: Token): Promise<Response<BalanceItemWithPayments[]>> => {
+        const request = Request.buildJson('GET', `/v${Version}/organization/members/${memberId}/balance`, organization.getApiHost());
+        request.headers.authorization = 'Bearer ' + token.accessToken;
+        return await testServer.test(memberBalanceEndpoint, request);
+    };
+
+    beforeAll(async () => {
+        period = await new RegistrationPeriodFactory({
+            startDate: new Date(2023, 0, 1),
+            endDate: new Date(2023, 11, 31),
+        }).create();
+
+        organization = await new OrganizationFactory({ period })
+            .create();
+
+        admin = await new UserFactory({
+            globalPermissions: Permissions.create({
+                level: PermissionLevel.Full,
+            }),
+            permissions: Permissions.create({
+                level: PermissionLevel.Full,
+            }),
+        })
```

Comment:
Probeer in tests altijd met users te werken die de absoluut minimum noodzakelijke access rights hebben, op die manier kunnen we beter fouten opsporen omdat we in development vaak met full permissions werken. In dit geval lijkt me het best om write access te hebben tot alle leden + financial write access.

Daarnaast misschien ook één specifieke test waarbij je bv. enkel toegang hebt tot één specifieke leeftijdsgroep en enkel daarvoor toch nog een aanrekening kan aanmaken als je ook toegang hebt tot financiele data van leden. + dan ook een negatieve test om te checken dat je 1. geen aanrekening kan aanmaken voor leden waar je geen toegnag tot hebt en 2. voor leden waarbij je wel toegang hebt maar geen financial write access